File: Swift-C++-IR-Comparison/test/structsAndClasses/Swift/Structs/StructsMemoryLayout.swift
@main
struct MyMain {

    static func main() {
        let res = Resolution()
        let explicitValues = Resolution(width: 32, height: 64, max: 3232)
    }
}

struct Resolution {
    var width: UInt32 = 0
    var height: UInt64 = 0
    var max: UInt32 = 0
}

File: AutoTranspiledTests/Swift-to-Cpp/try-1/StructsMemoryLayout.cpp

#include <iostream>

class Resolution {
	public:
		uint32_t width = 0;
		uint64_t height = 0;
		uint32_t max = 0;

		Resolution() {}

		Resolution(uint32_t w, uint64_t h, uint32_t m) {
			width = w;
			height = h;
			max = m;
		}
};

int main() {
	Resolution res;
	Resolution explicitValues(32, 64, 3232);

	return 0;
}